benches/bench_pyclass.rs

```diff
@@ -0,0 +1,133 @@
+#![feature(test)]
+
+extern crate test;
+use pyo3::prelude::*;
+use pyo3::PyIterProtocol;
+use test::Bencher;
+
+#[pyclass]
+pub struct MyClass {
+    #[pyo3(get, set)]
+    obj: PyObject,
+
+    #[pyo3(get, set)]
+    x: i32,
+
+    values: Vec<usize>,
+}
+
+#[pyclass]
+pub struct Iter {
+    inner: std::vec::IntoIter<usize>,
+}
+
+#[pyproto]
+impl PyIterProtocol for Iter {
+    fn __next__(mut slf: PyRefMut<'p, Self>) -> PyResult<Option<usize>> {
+        Ok(slf.inner.next())
+    }
+}
+
+#[pyproto]
+impl PyIterProtocol for MyClass {
+    fn __iter__(slf: PyRef<'p, Self>) -> PyResult<Py<Iter>> {
+        let iter = Iter {
+            inner: slf.values.clone().into_iter(),
+        };
+        PyCell::new(slf.py(), iter).map(Into::into)
+    }
+}
+
+#[bench]
+fn create_many_pyclass(b: &mut Bencher) {
+    b.iter(|| {
+        for i in 0..20 {
+            let gil = Python::acquire_gil();
+            let py = gil.python();
+            for j in 0..200 {
+                let _ = PyCell::new(
+                    py,
+                    MyClass {
+                        obj: py.None(),
+                        x: i * j,
+                        values: vec![],
+                    },
+                );
+            }
+        }
+    });
+}
+
+#[bench]
+fn iter_pyclass(b: &mut Bencher) {
+    let mut sum = 0;
+
+    b.iter(|| {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let obj = PyCell::new(
+            py,
+            MyClass {
+                obj: py.None(),
+                x: 0,
+                values: (0..10_000).into_iter().collect(),
```

Comment:
It looks that clippy complains about this.
```suggestion
                values: (0..10_000).collect(),
```